File: passenger/src/test/java/com/modsen/passenger/component/glue/StepDefinitions.java
package com.modsen.passenger.component.glue;

import com.modsen.passenger.dto.response.PassengerResponse;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import static org.assertj.core.api.Assertions.assertThat;


public class StepDefinitions {

    @Autowired
    private TestRestTemplate testClient;
    private Integer passengerId;
    private ResponseEntity<PassengerResponse> response;

    @Given("Client provides id {int}")
    public void clientProvidesId(Integer passengerId) {
        this.passengerId = passengerId;
    }

    @When("Client makes request to {string}")
    public void clientMakesRequestToApiPassengers(String endpoint) {
        response = testClient
                .getForEntity(endpoint + passengerId, PassengerResponse.class);
    }

    @Then("Should return OK status and expected passenger")
    public void shouldReturnOkStatus() {
        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
    }
}

File: kt-passenger/src/test/kotlin/com/modsen/ktpassenger/component/glue/StepDefinitions.kt
package com.modsen.ktpassenger.component.glue

import com.modsen.ktpassenger.dto.PassengerResponse
import io.cucumber.java.en.Given
import io.cucumber.java.en.Then
import io.cucumber.java.en.When
import org.assertj.core.api.Assertions
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity


class StepDefinitions(
    private val testClient: TestRestTemplate
) {

    private var passengerId: Int? = null
    private var response: ResponseEntity<PassengerResponse>? = null

    @Given("Client provides id {int}")
    fun clientProvidesId(passengerId: Int?) {
        this.passengerId = passengerId
    }

    @When("Client makes request to {string}")
    fun clientMakesRequestToApiPassengers(endpoint: String) {
        response = testClient
            .getForEntity(endpoint + passengerId, PassengerResponse::class.java)
    }

    @Then("Should return OK status and expected passenger")
    fun shouldReturnOkStatus() {
        Assertions.assertThat(response!!.statusCode).isEqualTo(HttpStatus.OK)
    }
}

